### PR TITLE
exec.cc: Fix for #580

### PR DESCRIPTION
**Pull request by su8**

### CODE REVIEW

**Comment by lasers** [comment]
@su8 This requires me to open identical PR. I don't want to do that because it'd keep making me an author of your fixes, lol. I want you to push out a new branch with better commit message like... `exec.cc: permit wrapping commands in quotes`.

I assume you need help? What do you have when you run `git remote -v`? And you can make new branch by issuing `git branch checkout -b su8-fixes-580`... and `git push upstream`.

**Comment by su8** [comment]
@lasers Wrapped commands should work now with the latest commit.

File: src/exec.cc [hunk 1]
```diff
@@ -55,6 +55,30 @@ struct execi_data {
 static FILE *pid_popen(const char *command, const char *mode, pid_t *child) {
   int ends[2];
   int parentend, childend;
+  char cmd[256];
```
[the remark this answers is not in the document]
Comment (su8) [reply src/exec.cc]: I don't know C++ to use std:string.

File: src/exec.cc [hunk 2]
```diff
@@ -55,6 +55,30 @@ struct execi_data {
...
+        skip = 1;
+        continue;
+      }
+    }
+    if ('\0' == *(str2+1) && 1 == skip) {
+      if (*str2 == '"' || *str2 == '\'') {
+        continue;
+      }
+    }
+    if (254 > x) {
```
[the remark this answers is not in the document]
Comment (su8) [reply src/exec.cc]: I used the length of the cmd char array length, will make a int variable that defines how long a char array should be, instead using p_max_size here.

**Comment by lasers** [comment]
Mister @su8, are you finished with the changes?

**Comment by su8** [comment]
Yes sir.

**Comment by lasers** [comment]
@su8... I probably should had mentioned this in first place.

Change commit from `exec.cc: Fix for #580` to `exec.cc: permit wrapping commands in quotes`.
Change description to `Closes #580`. 

Change title to `exec.cc: permit wrapping commands in quotes (#589)`  on squashing soon, I think.

**Comment by lasers** [comment]
Lol. Don't give me a heart attack. Do them. Also, check code smellz in my PR.

**Comment by su8** [comment]
@lasers code smells was addressed in the latest commit.

**Comment by brndnmtthws** [comment]
Merged #591

**Pull request closed without merging**